cmd/limactl/shell.go

```diff
@@ -91,40 +108,45 @@ func shellAction(clicontext *cli.Context) error {
 	logrus.Debugf("changeDirCmd=%q", changeDirCmd)
 
 	script := fmt.Sprintf("%s ; exec bash --login", changeDirCmd)
-	if clicontext.NArg() > 1 {
-		script += fmt.Sprintf(" -c %q", shellescape.QuoteCommand(clicontext.Args().Tail()))
+	if len(args) > 1 {
+		tailArgs := args[1:]
+		script += fmt.Sprintf(" -c %q", shellescape.QuoteCommand(tailArgs))
 	}
 
 	arg0, err := exec.LookPath("ssh")
 	if err != nil {
 		return err
 	}
 
-	args, err := sshutil.SSHArgs(inst.Dir, *y.SSH.LoadDotSSHPubKeys)
+	sshArgs, err := sshutil.SSHArgs(inst.Dir, *y.SSH.LoadDotSSHPubKeys)
 	if err != nil {
 		return err
 	}
 	if isatty.IsTerminal(os.Stdout.Fd()) {
 		// required for showing the shell prompt: https://stackoverflow.com/a/626574
-		args = append(args, "-t")
+		sshArgs = append(sshArgs, "-t")
 	}
-	args = append(args, []string{
+	sshArgs = append(sshArgs, []string{
 		"-q",
 		"-p", strconv.Itoa(inst.SSHLocalPort),
 		"127.0.0.1",
 		"--",
 		script,
 	}...)
-	cmd := exec.Command(arg0, args...)
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	logrus.Debugf("executing ssh (may take a long)): %+v", cmd.Args)
+	sshCmd := exec.Command(arg0, sshArgs...)
+	sshCmd.Stdin = os.Stdin
+	sshCmd.Stdout = os.Stdout
+	sshCmd.Stderr = os.Stderr
+	logrus.Debugf("executing ssh (may take a long)): %+v", sshCmd.Args)
 
 	// TODO: use syscall.Exec directly (results in losing tty?)
-	return cmd.Run()
+	err = sshCmd.Run()
+	if err != nil {
+		logrus.Warnf("shell command execute failed (%v)", err)
+	}
+	return nil
```

Comment:
I'm not sure about this change. It turns

```console
$ lima false
exit status 1
```

into

```console
$ lima false
WARN[0000] shell command execute failed (exit status 1)
```

Why is this preferable? I'm tempted to say that there should be no output at all; just the exit code needs to be propagated.

@AkihiroSuda Any opinion?

Reply:
I agree with @jandubois , we should retain the previous behavior.